hibernate-core/src/main/java/org/hibernate/query/SelectionQuery.java

```diff
@@ -291,7 +291,40 @@ default Stream<R> stream() {
 	 *
 	 * @since 6.3
 	 */
-	SelectionQuery<R> setEntityGraph(EntityGraph<R> graph, GraphSemantic semantic);
+	SelectionQuery<R> setEntityGraph(EntityGraph<? super R> graph, GraphSemantic semantic);
+
+	/**
+	 * Apply a {@linkplain jakarta.persistence.NamedEntityGraph named entity graph}
+	 * to the query.
+	 * <p>
+	 * May throw an {@link IllegalArgumentException} exception if no entity graph for a given name is found.
+	 *
+	 * @param graphName The name of the predefined named entity graph to apply
+	 * @param semantic The way an entity graph should be applied
+	 *
+	 * @since 7.0
+	 */
+	@Incubating
+	default SelectionQuery<R> setNamedEntityGraph(String graphName, GraphSemantic semantic) {
+		EntityGraph<? super R> namedEntityGraph = getNamedEntityGraph( graphName );
+		if ( namedEntityGraph == null ) {
+			throw new IllegalArgumentException( "No entity graph found for name " + graphName );
+		}
+		return setEntityGraph( namedEntityGraph, semantic );
+	}
```

Comment:
this felt like a more user-friendly way to apply that graph, but if you don't like it I'll drop it 😃

Reply:
I guess I'm not very keen on it because it pushes people toward using string-based names whereas we're actively trying to _discourage_ them from using strings.

What we _really_ want them to do is use `Entity_._NamedGraph` to get a typesafe reference to the named entity graph. We prefer that they _don't_ use these string-based lookups.

So, while I'm fine with adding _one_ new operation I'm not sure we want to bloat out the very important `SelectionQuery` interface with operations which only streamline the untypesafe usage.

Reply:
I guess I'm not very keen on it because it pushes people toward using string-based names whereas we're actively trying to discourage them from using strings.

What we really want them to do is use Entity_._NamedGraph to get a typesafe reference to the named entity graph. We prefer that they don't use these string-based lookups.

So, while I'm fine with adding one new operation I'm not sure we want to bloat out the very important SelectionQuery interface with operations which only streamline the untypesafe usage.

Reply:
🤷🏻 😃  it's just when I was adding that test example:
```java
SelectionQuery<Aa> query = s.createSelectionQuery( "from Aa", Aa.class );
query
    .setEntityGraph( 
        //  here:
        query.getNamedEntityGraph( "a-graph" ), GraphSemantic.FETCH )
    .getResultList();
```
above looks a bit odd to pass something to a query by getting it from the very same query.
IDK how often people generate the static metamodel in their projects, but if  "this is the way" 😃 I'm ok with reverting this bit 😃 just let me know

Reply:
> above looks a bit odd to pass something to a query by getting it from the very same query.

Yes I understand that.

> IDK how often people generate the static metamodel in their projects, but if "this is the way" 

It's the new way in JPA 3.2. We want to encourage them to use it.